src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -824,17 +824,22 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
      * @throws InterruptedException when interrupted
      */
     @NonNull
-    public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,?> build, FilePath workspace) throws IOException, InterruptedException {
+    public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,?> build, FilePath workspace, UnsupportedCommand cmd) throws IOException, InterruptedException {
```

Comment:
I don't think the existing signature should be deprecated since the argument, UnsupportedCommand, is only passed down by the Git Publisher. Hence it makes sense to retain the old signature method and add a new method which is used by the Git Publisher alone (earlier the CreateClient API was used commonly by both the SCM builder and Git Publisher)

Reply:
Unless there is a compelling reason to make the new method public, I'd rather leave it package protected so that only callers from within the git plugin can use it.  If you decide to keep it public, then the javadoc for the additional parameter needs to be made more general purpose so that other callers understand why they must provide a value there and how they can decide the correct value.